ports-of-call/portable_arrays.hpp

```diff
@@ -113,48 +182,21 @@ class PortableMDArray {
     return -1;
   }
 
-  // a function to get the total size of the array
   PORTABLE_FORCEINLINE_FUNCTION int GetSize() const {
-    return nx1_ * nx2_ * nx3_ * nx4_ * nx5_ * nx6_;
+    return std::accumulate(nxs_.cbegin(), nxs_.cend(), 1,
```

Comment:
Have you tested this on GPUs? I know Brendan had an issue and we had to write a custom accumulate. One could still use binary operators provided by the STL, but the accumulate itself wouldn't run. Perhaps we did something wrong, but I want to make sure this is still GPU callable.

Reply:
I would also be concerned about `std::accumulate`. Might be better to hardcode this one, or write a a recursive thing ourselves.

Reply:
It "worked" but gave warnings (even with `--expt-relaxed-constexpr`). I rewrote it as an explicit loop.